provider/pkg/provider/provider.go

```diff
@@ -676,9 +676,15 @@ func (k *kubeProvider) Configure(_ context.Context, req *pulumirpc.ConfigureRequ
 	if !k.clusterUnreachable {
 		config, err := kubeconfig.ClientConfig()
 		if err != nil {
-			k.clusterUnreachable = true
-			k.clusterUnreachableReason = fmt.Sprintf(
-				"unable to load Kubernetes client configuration from kubeconfig file: %v", err)
+			// Check if kubeconfig env var is set.
+			if set, _ := os.LookupEnv("KUBECONFIG"); set == "" {
```

Comment:
On second glance here this isn't checking to see if there's an explicit provider set

Reply:
The latest update doesn't perform this check anymore. The goal is to just point the user to a more authoritative source for configuration than what the kubernetes client returns.